---
output:
  html_document: default
  word_document: default
  pdf_document: default
---
# Data harmonisation

The goal of this exercise is to practise harmonising sets of snps, to get ready for an MR calculation. 

Load MendelianRandomization Package

```r
if (!require("pacman")) install.packages("pacman")
pacman::p_load(MendelianRandomization)
```


Matching up two or more sample sets of snps can be tricky. Things you need to remember

*Does each gene-outcome or gene-exposure set have consistent units?*

    1. Which is the effect snp?
    2. Match strand and direction of effect
    
*Check effect allele frequency*

## Set working directory
```r
setwd("~/Desktop/MRW")
```
## 1.1  Identify effect allele

Load and look at the three datasets provided for this exercise (Phenoscanner.csv, Paper_1.csv, Paper_2.csv).
Can you identify the effect allele column in each case?

```r
Paper1<-read.csv("Paper_1.csv", row=1)
Paper2<-read.csv("Paper_2.csv", row=1)
Pheno<-read.csv("Phenoscanner.csv", row=1)
head(Paper1)
head(Paper2)
head(Pheno)
```

What is the effect allele and non-effect allele of rs646776 based on information in paper 1 and paper 2?

How does information in pheno scanner help to determine which is the effect allele?


## 1.2 Palindromic snps

Identify the snps that are palindromic. Why do we need to be extra careful of these snps?

The Phenoscanner data includes strand data, but Paper 1 doesn’t. This is common. However we can use allele frequencies to match the direction of the snps. If both snps have A as their effect allele and the frequency of that allele is similar, then we can be confident they are aligned correctly.


If this is not the case, we need to either flip the direction (swap the effect allele and the reference over and multiply the beta by -1, replace EAF with 1- EAF) or change the strand (replace each allele with it’s DNA complement, change the strand label if one exists).
Note: this becomes more complicated if you are doing a MR analysis with a correlation matrix - you would need to think about how the correlation as well as the snps change direction.

## 1.2.2 Ancestry

First, identify which ancestry frequency column to use from the Phenoscanner data.

##  1.2.3

Put the pallindromic snps in Paper 1 into the same alignment as the Phenoscanner dataset.

## 1.3 Non-palindromic snps
Look at the non-palindromic snps in Paper 1 and compare them to the snps in the Phenoscanner set. 

Which ones are the right/wrong way round? 
Which are labelled on different strands?

### *When snps are missing, one option is to replace them with the results from proxy snps - snps highly in correlation with our original snp.*






